massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -903,15 +877,17 @@ impl ProtocolWorker {
         }
 
         if !new_operations.is_empty() {
-            // Add to pool, propagate when received outside of a header.
-            self.send_protocol_pool_event(ProtocolPoolEvent::ReceivedOperations {
-                operations: new_operations,
-                done_signal,
-            })
-            .await;
-
             // prune checked operations cache
             self.prune_checked_operations();
+
+            // Store operation, claim locally
+            let mut ops = self.storage.clone_without_refs();
+            ops.store_operations(new_operations.into_values().collect());
+            self.storage
+                .claim_operation_refs(ops.get_local_operation_ids());
```

Comment:
Why do you need to claim them locally? Could you not simply have `note_operations_..` return a storage which owns the newly processed operations, and send this on a message to consensus?

Reply:
@gterzian here is a doc: https://github.com/massalabs/massa/pull/2901